app/src/main/java/com/soen490chrysalis/papilio/view/UserProfileActivity.kt

```diff
@@ -83,4 +366,49 @@ class UserProfileActivity : AppCompatActivity()
         }
         return super.onOptionsItemSelected(item)
     }
+
+    fun LinearLayoutCreator(viewsToAdd: Array<View>): LinearLayout {
+        val params = LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.MATCH_PARENT
+        )
+
+        val subViewParams = LayoutParams(
+            LinearLayout.LayoutParams.MATCH_PARENT,
+            LinearLayout.LayoutParams.WRAP_CONTENT
+        )
+
+        val newLayout = LinearLayout(this)
+        newLayout.layoutParams = params
+        for (view in viewsToAdd) {
+            view.layoutParams = subViewParams
+            newLayout.addView(view)
+        }
+        newLayout.orientation = LinearLayout.VERTICAL
+        newLayout.setPadding(50, 75, 50, 75)
+
+        return newLayout
+    }
+
+    fun DialogBuilderCreator(title: String, layout: LinearLayout): AlertDialog.Builder {
+        val builder = AlertDialog.Builder(this)
+        builder.setTitle(title)
+        builder.setView(layout)
+        builder.setPositiveButton(
+            "Save",
+            DialogInterface.OnClickListener { dialog_: DialogInterface, _: Int ->
```

Comment:
I don't know if you are developing on android studio but it sometimes suggests to remove redundant variables/statements if you hover over those warnings like this:
![image](https://user-images.githubusercontent.com/64218770/211125626-37152ae1-574a-42b7-a1ef-e6c945e0ddac.png)

I think you can remove these redundant keywords / unused variables